src/MoBi.Presentation/Tasks/SimulationUpdateTask.cs

```diff
@@ -112,18 +112,14 @@ private ICommand<IMoBiContext> updateSimulation(
          //create model using referencing templates
          var results = _simulationFactory.CreateModelAndValidate(simulationConfigurationReferencingTemplates, simulationToUpdate.Model.Name, message);
 
-         var simulationBuildConfiguration = createSimulationConfigurationToUseInSimulation(simulationConfigurationReferencingTemplates);
+         //create a clone then that will be saved in the simulation
+         var simulationBuildConfiguration = _cloneManager.Clone(simulationConfigurationReferencingTemplates);
```

Comment:
for me this indirection used once calling a method that is clear does not make sense. It just adds complexity. I could not tell it was a clone at first sight